pkg/deploy/internal/setting/setting_test.go

```diff
@@ -151,7 +151,7 @@ func TestDeploySetting_ManagementZone_FailToDecodeMZoneID(t *testing.T) {
 		Parameters: testutils.ToParameterMap(parameters),
 	}
 	props := map[string]interface{}{"scope": "environment"}
-	resolvedEntity, err := Deploy(context.TODO(), c, props, "", conf, "")
+	resolvedEntity, err := Deploy(t.Context(), c, props, "", conf)
```

Comment:
nit: I see you need to update the signature of the Deploy and are using the new `t.Context()` here but not in the changed methods on line 73, 85, 104, 1029. I suspect the others will be updated in a separate PR?

Reply:
Since we have multiple PRs open changing the context in tests, I would do minimal changes right now. I will even revert this change, and we swoop in later to properly do it.